### ISSUE TITLE
Loading multiple GLB files with animations fails

### ISSUE THREAD

**Issue by boonto** [jMonkeyEngine/jmonkeyengine#1003]
I ran into this issue when loading a simple GLB file first and then loading another one with an animation.
When loading the animation GLB file second, all of the `TransformTrack` data was scrambled.

After looking into it, I found that the `bufferIndex` when reading the data is wrong, which causes the `GlbLoader` to use the buffer of the previously loaded GLB file.

**Comment by stephengold** [comment]
Could you provide any more details about your findings?

**Comment by boonto** [comment]
Sure, it was late yesterday.  Here is a minimal example:
```
import com.jme3.anim.AnimComposer;
import com.jme3.app.SimpleApplication;
import com.jme3.math.ColorRGBA;
import com.jme3.scene.Spatial;

public class Main extends SimpleApplication {
    public static void main(String[] args) {
        Main app = new Main();
        app.setShowSettings(false);
        app.start();
    }

    @Override
    public void simpleInitApp() {
        this.viewPort.setBackgroundColor(ColorRGBA.DarkGray);

//        Spatial cube = this.assetManager.loadModel("cube.glb");
//        Spatial monkey = this.assetManager.loadModel("monkey.glb");
        Spatial anim = this.assetManager.loadModel("sway.glb");

        anim.breadthFirstTraversal(spatial -> {
            AnimComposer control = spatial.getControl(AnimComposer.class);
            if (control != null) {
                control.getAnimClipsNames()
                       .stream()
                       .findFirst()
                       .ifPresent(control::setCurrentAction);
            }
        });

        this.rootNode.attachChild(anim);
    }
}
```
This snippet works.
But when the cube/monkey is loaded first the program crashes, due to looking for the animations in the previously loaded model's data.
When I encountered it yesterday it even successfully loaded the data and produced wrong animations.

Here are the models: [resources.zip](https://github.com/jMonkeyEngine/jmonkeyengine/files/2740333/resources.zip)

**Comment by stephengold** [comment]
At the start of each `loadModel()`, the `GltfLoader`'s `dataCache` is cleared and a new `animations` array is instantiated. I don't see what data are being reused. Could you clarify for me?

**Comment by boonto** [comment]
I don't think thats the case for the `GlbLoaders`'s [`data`](https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/GlbLoader.java#L17). This is what gets accessed by the `readData` method in the end.

**Comment by stephengold** [comment]
Now I get it. That explains why the issue only occurs with `.glb`, not `.gltf`. I'll add a `data.clear()` to address the immediate issue.

> I don't think thats the case for the `GlbLoaders`'s [`data`](https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/GlbLoader.java#L17). This is what gets accessed by the `readData` method in the end.

**Comment by stephengold** [comment]
The fix is integrated into the `master` branch at GitHub. @boonto please test.

**Comment by boonto** [comment]
Works!

**Comment by stephengold** [comment]
Fixed in `master` at commit hash 34b32bf29b3a1c675fe30442e089fcc1de164692 .

Lets make this a high priority for the next JME release.
